File: C/Examples1/practice1_1.c
/*
 * practice1_1.c
 *
 * プログラミングBI 演習課題1-1
 *
 * Copyright (C) 2007-2026, S. Muramatsu
 *
 */
#include <stdio.h>

int main()
{
	/* 変数宣言 */
	double x; /* 独立変数 */
	double y; /* 従属変数 (関数の近似値) */
	double z; /* xの二乗の変数*/
	int n; /* 近似次数 */
	int i; /* 繰り返し変数の宣言 */
	double f; /* 階乗値 */
	double p; /* x の累乗値 */
	double q; /* 各項の値 */
	double r; /* 各項の符号を表す*/
	/* データの読み込み(倍精度, 整数) */
	scanf("%lf%d",&x,&n);

	/* 近似値の初期化 */
	y = 1.0;
	f = 1.0;
	p = 1.0;
	r = -1.0;
	z = x*x; 
	for (i=1; i<=n; i++) {
		f = f * (2 * (double)i - 1); /* f *= 2 * (double)i - 1; とも書ける */
		f = f * (2 * (double)i); /* f *= 2 * (double)i; とも書ける */
		p = p * z; /* p *= z; とも書ける */
		q = p/f; /* 各項を求めた */
		y = y + r * q; /* 各項の値を加えていく */
		r = -r; /* 各項の符号の調整 */
	}

	/* e^x の近似値の表示 */
	printf("マクローリン展開による次数 %d までの cos(%g) の近似値\n"
		"    =%21.16g\n", n, x, y);
	
	return 0;
}

File: Python/Examples1/practice1_1.py
"""
practice1_1.py
(c) Copyright 2022-2025, Shogo MURAMATSU, All rights reserved
"""
import sys
import math

# 電源電圧値
SRC_VOLTAGE = 5.0
# 要素の数
NUM_ELEMENTS = 4

def main(args):
    if len(args) != 5:
        print('エラー：引数の数' + str(len(args)))
        sys.exit(0)

    # 要素それぞれの格納
    elements = [ 0.0 for idx in range(NUM_ELEMENTS) ]
    for idx in range(NUM_ELEMENTS):
        elements[idx] = float(args[idx+1])
        
    # 抵抗Rに流れる電流の大きさ|I|の計算
    E = SRC_VOLTAGE
    R, L, C, ω = elements[0], elements[1], elements[2], elements[3]
    I = E / math.sqrt(R**2 + (ω * L - 1 / (ω * C))**2)

    # 入力した値と電流の大きさ|I|の表示
    print('電源電圧 |E| = ' + str(SRC_VOLTAGE) + ' V')
    print('抵抗 R = ' + str(R) + ' Ω')
    print('インダクタ L = ' + str(L) + ' H')
    print('コンデンサ C = ' + str(C) + ' F')
    print('角周波数 ω = ' + str(ω) + ' rad/s')
    print('Rに流れる電流 |I| = ' + str(I) + ' A')

if __name__ == '__main__':
    main(sys.argv)
        
        
